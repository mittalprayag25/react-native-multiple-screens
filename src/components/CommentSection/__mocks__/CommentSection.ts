import {createElement} from 'react';

jest.mock('..', () => createElement('CommentSection'));
