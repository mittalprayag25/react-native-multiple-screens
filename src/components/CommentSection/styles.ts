import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  commentColor: {
    color: 'grey',
    fontSize: 15,
  },
  subComment: {
    paddingHorizontal: 10,
    marginVertical: 2,
    backgroundColor: 'rgb(246, 246, 239)',
  },
  by: {
    fontWeight: 'bold',
    fontSize: 16,
  },
});

export default styles;
