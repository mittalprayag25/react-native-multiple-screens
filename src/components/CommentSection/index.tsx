import React, {memo} from 'react';
import {View, Text} from 'react-native';
import {parseSpecialCharacters} from './../../utils/utilities';

import styles from './styles';

const CommentSection = ({comments, data}: any): JSX.Element => {
  const getComments = kids => {
    const arr = [];
    kids.forEach((item: any) => {
      if (data[item]) {
        arr.push(data[item]);
      }
    });
    return arr;
  };

  const ui = (obj: any) => {
    return (
      <View style={styles.subComment}>
        {obj.by && (
          <Text style={styles.by}>
            -{'>'} ({obj.by})
          </Text>
        )}
        {obj.text && (
          <Text style={styles.commentColor}>
            {parseSpecialCharacters(obj.text)}
          </Text>
        )}
        {obj.kids && comment(obj.kids)}
      </View>
    );
  };

  const comment = (dat: any) => {
    return getComments(dat).map(obj => {
      return ui(obj);
    });
  };

  return comments && <View>{comment(comments)}</View>;
};

export default memo(CommentSection);
