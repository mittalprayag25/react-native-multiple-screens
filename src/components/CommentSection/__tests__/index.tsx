import {render} from '@testing-library/react-native';
import React from 'react';

import CommentSection from '..';

describe('CommentSection ', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    data: {
      30821604: {
        by: 'cute_boi',
        id: 30821604,
        kids: [30821923],
        parent: 30820894,
        text: 'comment text',
        time: 1648398990,
        type: 'comment',
      },
      30821923: {
        by: 'commentNested1',
        id: 30821604,
        kids: [30821944],
        parent: 30820894,
        text: 'comment text',
        time: 1648398990,
        type: 'comment',
      },
      30821944: {
        by: 'commentNested2',
        id: 30821604,
        kids: [30821922],
        parent: 30820894,
        text: 'comment text',
        time: 1648398990,
        type: 'comment',
      },
      30820812: {
        by: 'newComment',
        id: 30821604,
        kids: [30821922],
        parent: 30820894,
        text: 'comment text',
        time: 1648398990,
        type: 'comment',
      },
    },
    comments: [30821604, 30820812],
  };

  it('should match snapshot', () => {
    const tree = render(<CommentSection {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('CommentSection when feeds not downloaded', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    data: {},
    comments: [30821604, 30820812],
  };

  it('should match snapshot', () => {
    const tree = render(<CommentSection {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('CommentSection when no comments', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    data: {},
    comments: [],
  };

  it('should match snapshot', () => {
    const tree = render(<CommentSection {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
