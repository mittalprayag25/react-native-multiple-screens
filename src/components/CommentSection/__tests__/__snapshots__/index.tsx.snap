// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentSection  should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(246, 246, 239)",
        "marginVertical": 2,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      -
      &gt;
       (
      cute_boi
      )
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
        }
      }
    >
      comment text
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "rgb(246, 246, 239)",
          "marginVertical": 2,
          "paddingHorizontal": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        -
        &gt;
         (
        commentNested1
        )
      </Text>
      <Text
        style={
          Object {
            "color": "grey",
            "fontSize": 15,
          }
        }
      >
        comment text
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "rgb(246, 246, 239)",
            "marginVertical": 2,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          -
          &gt;
           (
          commentNested2
          )
        </Text>
        <Text
          style={
            Object {
              "color": "grey",
              "fontSize": 15,
            }
          }
        >
          comment text
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgb(246, 246, 239)",
        "marginVertical": 2,
        "paddingHorizontal": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      -
      &gt;
       (
      newComment
      )
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
        }
      }
    >
      comment text
    </Text>
  </View>
</View>
`;

exports[`CommentSection when feeds not downloaded should match snapshot 1`] = `<View />`;

exports[`CommentSection when no comments should match snapshot 1`] = `<View />`;
